webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/pdf/GeneralScrambleSheet.kt

```diff
@@ -80,43 +74,20 @@ class GeneralScrambleSheet(scrambleSet: ScrambleSet, activityCode: ActivityCode)
-
-        val longestScramble = scrambles.maxBy { it.length } ?: ""
-
-        val longestAlignedScramble = scrambles.map { StringUtil.padTurnsUniformly(it, WIDEST_CHAR_STRING) }
-            .maxBy { it.length } ?: longestScramble
+    private fun getFontConfiguration(availableArea: Rectangle, scrambles: List<String>): Font {
+        val longestScramble = scrambles.flatMap { it.split(NEW_LINE) }.maxBy { it.length } ?: ""
+        val maxLines = scrambles.map { it.split(NEW_LINE) }.map { it.count() }.max() ?: 1
 
-        val longestScrambleOneLine = "\n" !in longestAlignedScramble//Masked
+        val fontSize = PdfUtil.fitText(Font(FontUtil.MONO_FONT), longestScramble, availableArea, FontUtil.MAX_SCRAMBLE_FONT_SIZE, true) // FIXME const
+        val fontSizeIfIncludingNewlines = availableArea.height / maxLines
 
-        // I don't know how to configure ColumnText.fitText's word wrapping characters,
-        // so instead, I just replace each character I don't want to wrap with M, which
-        // should be the widest character (we're using a monospaced font,
-        // so that doesn't really matter), and won't get wrapped.
-        val longestScrambleMasked = longestScramble.replace(".".toRegex(), WIDEST_CHAR_STRING)
-        val longestAlignedScrambleMasked = longestAlignedScramble.replace("\\S".toRegex(), WIDEST_CHAR_STRING)
-        val longestAlignedScrambleMaskedAndStuffed = longestAlignedScrambleMasked.replace(" ".toRegex(), WIDEST_CHAR_STRING)
-
-        val fontSizeForMaskedUnaligned = PdfUtil.fitText(Font(FontUtil.MONO_FONT), longestScrambleMasked, availableArea, FontUtil.MAX_SCRAMBLE_FONT_SIZE, false, 1f) // FIXME const
-
-        // If the scramble contains newlines, then we *only* allow wrapping at the newlines.
-        val longestRespectingNewlines = longestAlignedScrambleMasked.takeIf { longestScrambleOneLine }
-            ?: longestAlignedScrambleMaskedAndStuffed
-        val fontSizeIfIncludingNewlines = PdfUtil.fitText(Font(FontUtil.MONO_FONT), longestRespectingNewlines, availableArea, FontUtil.MAX_SCRAMBLE_FONT_SIZE, true, 1f) // FIXME const
-
-        val oneLine = longestScrambleOneLine && fontSizeForMaskedUnaligned >= FontUtil.MINIMUM_ONE_LINE_FONT_SIZE
-        val perfectFontSize = fontSizeForMaskedUnaligned.takeIf { oneLine } ?: fontSizeIfIncludingNewlines
+        val perfectFontSize = min(fontSize, fontSizeIfIncludingNewlines)
 
         return Font(FontUtil.MONO_FONT, perfectFontSize, Font.NORMAL)
     }
 
     private fun requiresHighlighting(scrambleColumnWidth: Float, scrambleFont: Font, scrambles: List<String>): Boolean {
-        val paddedScrambles = scrambles.map { StringUtil.padTurnsUniformly(it, PdfUtil.NON_BREAKING_SPACE.toString()) }
-        val lineChunks = paddedScrambles.map { it.splitToLineChunks(scrambleFont, scrambleColumnWidth) }
+        val lineChunks = scrambles.map { it.splitToLineChunks(scrambleFont, scrambleColumnWidth) }
```

Comment:
You moved the padding to within the `splitToLineChunks` method. Did you check that this refactor doesn't conflict with the behaviour of other PDF classes that use `splitToLineChunks` (if any)?

Reply:
Every single time split lines was called, there was a padding before. I checked and that is completely OK.

Reply:
I just remembered. There was a place that splitted scrambles without the padTurnsUniformly. This caused wrong calculation of the fontsize. Font size was calculated in a way (using splitlines) and when things would get rendered, splitlines worked differently. It took me forever to find, it was very frustrating. There were fireworks when I realized this.